src/index.js

```diff
@@ -0,0 +1,147 @@
+  cityNameLabel.innerText = 'City Name';
+});
+
+gettemperature.addEventListener('click', () => {
+  findLocation(currCity);
+});
+// --------------------------------------wave4 Calling APIs-------------------
+const findLocation = (query) => {
+  axios
+    .get('http://127.0.0.1:5000/location', {
+      params: {
+        q: query,
+        format: 'json',
+      },
+    })
+    .then((response) => {
+      latitude = response.data[0].lat;
+      longitude = response.data[0].lon;
+      console.log(response);
+      console.log('success in findLatitudeAndLongitude!', latitude, longitude);
+      // make the next API call here!
+      // findcity(latitude, longitude);
+      getWeather(latitude, longitude);
+    })
+    .catch((error) => {
+      console.log('error in findLatitudeAndLongitude!');
+    });
+};
+
+const getWeather = (latitude, longitude) => {
+  axios
+    .get('http://127.0.0.1:5000/weather', {
+      params: {
+        format: 'json',
+        lat: latitude,
+        lon: longitude,
+      },
+    })
+    .then((response) => {
+      currentTemp = Number(response.data.main.temp);
+      count = Math.floor((currentTemp - 273.15) * 1.8) + 32;
+      totalCount.innerText = count;
+      console.log('success in find temperature!', response.data.main.temp);
+    })
+    .catch((error) => {
+      console.log('error in weather!');
+    });
+};
+
+// --------------------------------------wave5 Selecting the Sky-------------------
+const selectElement = document.getElementById('slide');
+const skyPicture = document.getElementById('skyPicture');
+
+const changeselectionSky = (value) => {
+  if (value === 'Sunny') {
+    skyPicture.textContent = '☁️ ☁️ ☁️ ☀️ ☁️ ☁️';
+  } else if (value === 'Cloudy') {
+    skyPicture.textContent = '☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️';
+  } else if (value === 'Rainy') {
+    skyPicture.textContent = '🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧';
+  } else if (value === 'Snowy') {
+    skyPicture.textContent = '🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨';
+  }
+};
+selectElement.addEventListener('change', (event) => {
+  // console.log(event.target);
+  const value = event.target.value;
+  // console.log(event.target.value);
+  changeselectionSky(value);
+});
+// --------------------------------wave2  Increase and Decrease Temperature----------------------
+
+// Select increment and decrement buttons
+const increaseElement = document.getElementById('up');
+const decreaseElement = document.getElementById('down');
+// Select total count
+const totalCount = document.getElementById('totalCount');
+// Variable to track count
+// let count = 49;
+// Display initial count value
+// totalCount.innerHTML = count;
+// Function to increment/decrement count
+
+// Function to increment/decrement count
+const handleIncrease = () => {
+  count++;
+  totalCount.innerHTML = count;
+  checkTemperatureRange(Number(totalCount.innerText));
+};
+
+const handleDecrease = () => {
+  count--;
+  totalCount.innerHTML = count;
+  checkTemperatureRange(Number(totalCount.innerText));
+};
```

Comment:
These work well! Can you consider how you could make your code more DRY by combining them into a single function?